tests/future/batch_semaphore.rs

```diff
@@ -0,0 +1,654 @@
+use crate::basic::clocks::{check_clock, me};
+use futures::stream::FuturesUnordered;
+use futures::{FutureExt, StreamExt};
+use shuttle::future::{self, batch_semaphore::*};
+use shuttle::{check_dfs, check_random, current, thread};
+use std::collections::HashSet;
+use std::sync::atomic::{AtomicUsize, Ordering};
+use std::sync::Arc;
+use std::sync::Mutex;
+use test_log::test;
+
+#[test]
+fn batch_semaphore_basic() {
```

Comment:
nit: Do we need the `batch_semaphore` prefix for all of the tests — they are in the `batch_semaphore.rs` file after all.